### PR TITLE
Port analytical LBA4 likelihood to JAX

### PR DESCRIPTION
**Pull request by krishnbera**
## Summary

Port HSSM's analytical LBA4 likelihood from PyTensor to native JAX while preserving the existing lba4 model interface.

Changes:
- implement JAX LBA4 density/CDF and four-choice log-likelihood evaluation
- support JIT execution, trialwise vector parameters, autodiff, and JAX/PyTensor bridging
- configure the LBA4 analytical likelihood with backend="jax"
- add direct numerical/gradient/configuration coverage
- add a deterministic one-chain NumPyro LBA4 sampling smoke test

## Validation

- MPLCONFIGDIR=/tmp/.mpl PYTENSOR_FLAGS=compiledir=/tmp/hssm-pytensor-lba4 .venv/bin/pytest tests/test_likelihoods_lba.py tests/test_modelconfig.py tests/distribution_utils/test_distribution_utils.py tests/distribution_utils/test_jax.py --no-cov -q — 62 passed
- marked slow LBA sampling test via -m slow — passed
- uv run pyrefly check — 0 errors
- Ruff production check and format check — passed
- git diff --check — passed

The checkout does not register the older --runslow pytest option, so the slow test was selected with -m slow. Existing sampler/NumPyro warnings and tiny-run divergence warnings remain expected for the 10-draw smoke run.

## Summary by CodeRabbit

* **New Features**
  * Added JAX support for the four-choice LBA likelihood.
  * Enabled JIT compilation, vectorized inputs, and parameter gradient calculations.
  * Updated the default four-choice LBA configuration to use the JAX backend.

* **Bug Fixes**
  * Invalid parameter configurations now return an appropriate invalid likelihood result.

* **Tests**
  * Added coverage for four-choice LBA sampling and JAX likelihood behavior.

### CODE REVIEW

**Review by AlexanderFengler — APPROVED** [review]
this looks fine to me. Thanks @krishnbera
